alacritty_terminal/src/tty/unix.rs

```diff
@@ -152,9 +152,9 @@ pub fn new<C>(config: &Config<C>, size: &SizeInfo, window_id: Option<usize>) ->
         let shell_name = pw.shell.rsplit('/').next().unwrap();
         let argv = vec![String::from("-c"), format!("exec -a -{} {}", shell_name, pw.shell)];
 
-        Shell::new_with_args("/bin/bash", argv)
+        Shell::new_with_args(String::from("/bin/bash"), argv)
     } else {
-        Shell::new(pw.shell)
+        Shell::new(String::from(pw.shell))
```

Comment:
Having the parameter take `Into<String>` as suggested earlier would make it much easier to use.

Reply:
True, but that is the price of specificity - reducing generality in the interest of simplicity. I thought it was a good trade, given how short the list of useful `Into<String>` producers for this use case is. I suppose the benefit is less evident without the larger intent in view: I'd like to see all the config facilities centralized, because at present they're pretty scattered. I've found practical simplification of hypothetical generality helpful in that endeavor. But I certainly don't want to waste your time, as I did with the 2018 edition font mod episode, so let me know if I'm heading in the wrong direction here and I'll close this pull and begin work on a different way.

Reply:
I don't see how this simplifies things.

With what we currently have, every consumer of this API needs to convert their input to Strings, adding at minimum `.into()` to every single call.

While using `Into<String>` as parameter allows passing both `&str` and `String` without any disadvantages, while doing all the work in the callee, rather than the caller.

If you ask me, removing duplicates from the callsite and moving it into the callee is simplicity, not duplicating everything.

> But I certainly don't want to waste your time, as I did with the 2018 edition font mod episode, so let me know if I'm heading in the wrong direction here and I'll close this pull and begin work on a different way.

Please don't make any rash decisions. Fundamentally I don't see anything wrong with this PR, but I cannot merge something unless it is implemented properly and following all the repositorie's guidelines.

Reply:
> I don't see how this simplifies things.

An unneeded type (Cow), lifetime (static), trait bounds, and trait implementation was removed. The cost is a more specific parameter being passed from the OS implementation file.

> Please don't make any rash decisions.

I'm not, it isn't as if I take your objections personally. I just know, through prior experience with your reviews, that when it starts getting into matters of opinion - the probability for a merge drops very fast. So the sooner that can be shortcut, the better.

Reply:
> An unneeded type (Cow), lifetime (static), trait bounds, and trait implementation was removed.

I was talking about taking a `String` rather than `Into<String>`, not this PR in general.

Reply:
> ...when your opinions are different than the guidelines of the Alacritty project, there's no chance of them being merged.

I've got no problem with the benevolent dictator model, I'd actually say that it is preferable in most software projects (including this one)... so I don't understand why you'd suggest that there is some kind of codified guideline that has been violated (here or in the prior event that I referenced). I see no such guideline, and I see nothing in my conduct suggesting that I operate under the illusion that a project guideline related to this even exists.

I'm going to assume that your final sentiment on standards was directly related to the code under discussion - and the short circuit I was asking for.

Reply:
> I see no such guideline, and I see nothing in my conduct suggesting that I operate under the illusion that a project guideline related to this even exists.

That's why I'm telling you. Of course not every little standard is documented, but I do know how it should be done to be in line with the rest of the Alacritty codebase. Which in this case would be `Into<String>` and not `String`.